### ISSUE TITLE
picodet训练出错

### ISSUE THREAD

**Issue by 205418367** [PaddlePaddle/PaddleDetection#7029]
### 问题确认 Search before asking

- [X] 我已经查询[历史issue](https://github.com/PaddlePaddle/PaddleDetection/issues)，没有发现相似的bug。I have searched the [issues](https://github.com/PaddlePaddle/PaddleDetection/issues) and found no similar bug report.

### Bug组件 Bug Component

Training

### Bug描述 Describe the Bug

问题描述：
我用的是自己制作的数据集，训练到100个epoch时就出错，训练用的图片尺寸为1280*1280。
执行命令：
export  CUDA_VISIBLE_DEVICES=0,1
python -m paddle.distributed.launch --gpus 0,1 tools/train.py -c configs/picodet/picodet_l_640_coco_lcnet.yml

[09/27 11:21:53] ppdet.utils.checkpoint INFO: Save checkpoint: output/picodet_l_640_coco_lcnet
[09/27 11:22:01] ppdet.engine INFO: Epoch: [100] [ 0/13] learning_rate: 0.024642 loss_vfl: 0.686281 loss_bbox: 0.445803 loss_dfl: 0.386246 loss: 1.544325 eta: 2:03:02 batch_cost: 2.5983 data_cost: 1.4361 ips: 3.0789 images/s
Error: /paddle/paddle/fluid/operators/gather.cu.h:62 Assertion `index_value >= 0 && index_value < input_dims[j]` failed. The index is out of bounds, please check whether the dimensions of index and input meet the requirements. It should be less than [3] and greater than or equal to 0, but received [0]
Error: /paddle/paddle/fluid/operators/gather.cu.h:62 Assertion `index_value >= 0 && index_value < input_dims[j]` failed. The index is out of bounds, please check whether the dimensions of index and input meet the requirements. It should be less than [3] and greater than or equal to 0, but received [0]
Error: /paddle/paddle/fluid/operators/gather.cu.h:62 Assertion `index_value >= 0 && index_value < input_dims[j]` failed. The index is out of bounds, please check whether the dimensions of index and input meet the requirements. It should be less than [3] and greater than or equal to 0, but received [0]
Error: /paddle/paddle/fluid/operators/gather.cu.h:62 Assertion `index_value >= 0 && index_value < input_dims[j]` failed. The index is out of bounds, please check whether the dimensions of index and input meet the requirements. It should be less than [3] and greater than or equal to 0, but received [0]
Error: /paddle/paddle/fluid/operators/gather.cu.h:62 Assertion `index_value >= 0 && index_value < input_dims[j]` failed. The index is out of bounds, please check whether the dimensions of index and inpu

### 复现环境 Environment

ubuntu 18.04
cuda-11.2
paddlepaddle-gpu==2.2.2
release/2.5

### Bug描述确认 Bug description confirmation

- [X] 我确认已经提供了Bug复现步骤、代码改动说明、以及环境信息，确认问题是可以复现的。I confirm that the bug replication steps, code change instructions, and environment information have been provided, and the problem can be reproduced.

### 是否愿意提交PR？ Are you willing to submit a PR?

- [ ] 我愿意提交PR！I'd like to help by submitting a PR!

**Comment by lyuwenyu** [comment]
中间mAP指标正常嘛， 另外建议使用最新的paddle版本 2.3.2

**Comment by 205418367** [comment]
mAP指标正常,训练出来的模型可以正常检测

**Comment by 205418367** [comment]
这个steps设置多少比较合适？跟数据集有关系吧？
LearningRate:
  base_lr: 0.03
  schedulers:
  - !CosineDecay
    max_epochs: 100
  - !LinearWarmup
    start_factor: 0.1
    steps: 20

**Comment by 205418367** [comment]
已经解决，num_class的数量不对导致训练到100个epoch时出错！
这个问题能解答一下吗？
这个steps设置多少比较合适？跟数据集有关系吧？
LearningRate:
base_lr: 0.03
schedulers:

!CosineDecay
max_epochs: 100
!LinearWarmup
start_factor: 0.1
steps: 20
